config/configcompression/compressiontype.go

```diff
@@ -24,18 +34,25 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
+func (ct *TypeWithLevel) UnmarshalText() (TypeWithLevel, error) {
```

Comment:
The function signature should not be changed. We need to keep `encoding.TextUnmarshaler` interface implementation

Reply:
Yes made the change.